### PR TITLE
Updating project dependencies

### PR DESCRIPTION
**Pull request by gleopoldo**
I've made a few improvements in this PR:

- [x] Added an CLI to setup development environment using docker, this way any developer who wants to contribute to the project just need docker as dependency, instead of having several ruby versions installed locally;
- [x] Upgraded all dependencies to their respective supported versions;
- [x] Added supports for newer ruby versions;

I'll leave activerecord upgrade to `6.0.0` for another PR for now

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -18,3 +18,5 @@
 mkmf.log
 *.sqlite3
 *.gem
+.gems/
+Gemfile.lock
```
Comment (fabioperrella) [inline .gitignore:22]: nice!

File: scripts/heartcheck [hunk 2]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash -e
```
Comment (fabioperrella) [inline scripts/heartcheck:1]: What about mentioning this script in README?

File: lib/heartcheck/activerecord/version.rb [hunk 3]
```diff
@@ -2,6 +2,6 @@
 module Heartcheck
   # gem version
   module Activerecord
-    VERSION = '0.0.3'
+    VERSION = '0.1.0'.freeze
```
Comment (fabioperrella) [inline lib/heartcheck/activerecord/version.rb:5]: What about adding the change in CHANGELOG in this MR also?

File: scripts/heartcheck (hunk 2 above)
Comment (gleopoldo) [reply scripts/heartcheck:1]: Nice one! I've added it

File: .gitignore (hunk 1 above)
Comment (gleopoldo) [reply .gitignore:22]: Thanks! :+1:

File: lib/heartcheck/activerecord/version.rb (hunk 3 above)
Comment (gleopoldo) [reply lib/heartcheck/activerecord/version.rb:5]: I haven't found a CHANGELOG file in this project, so I'm not sure about how new version updates are being documented... Anyway, I'm going to add a CHANGELOG file to the project using the [Keep a Changelog](https://keepachangelog.com/pt-BR/1.0.0/) format. Can you take a look?

**Pull request merged**